src/gluonts/nursery/duru/data.py

```diff
@@ -0,0 +1,488 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+import gluonts.dataset
+import torch
+import torch as T
+import numpy as np
+import sklearn.model_selection
+import pickle
+import os
+import torchvision.transforms as transforms
+from gluonts.dataset.repository.datasets import get_dataset, dataset_recipes
+from gluonts.dataset.split import OffsetSplitter
+from utils import calculate_padding_base_2
+from gluonts.dataset.rolling_dataset import StepStrategy, NumSplitsStrategy
+from gluonts.dataset.rolling_dataset import generate_rolling_dataset
+
+
+def set_up_data(H):
+    # TODO split into train/val/test
+
+    if H.dataset == "dummy":
+        n_samples_train_val = 1000
+        n_samples_val = 200
+        n_samples_test = 500
+        # -
+        H.n_meas = 5
+        n_steps = 10000
+
+        # Version 1): random
+        X_train = (
+            torch.empty(n_samples_train_val, H.n_meas, n_steps)
+            .normal_()
+            .float()
+            .numpy()
+        )  # Conv1D expects (N,C_in,L_in), where C_in=number of channels=measurement dimension, L_in=length of signal sequence
+        Y_train = torch.empty(n_samples_train_val).bernoulli_().float().numpy()
+        X_test = (
+            torch.empty(n_samples_test, H.n_meas, n_steps)
+            .normal_()
+            .float()
+            .numpy()
+        )  # Conv1D expects (N,C_in,L_in), where C_in=number of channels=measurement dimension, L_in=length of signal sequence
+        Y_test = torch.empty(n_samples_test).bernoulli_().float().numpy()
+        # Version 2): all ones
+        # X_train = torch.ones(n_samples_train_val, H.n_meas, n_steps).float().numpy()  # Conv1D expects (N,C_in,L_in), where C_in=number of channels=measurement dimension, L_in=length of signal sequence
+        # Y_train = torch.ones(n_samples_train_val).float().numpy()
+        # X_test = torch.ones(n_samples_test, H.n_meas, n_steps).float().numpy()  # Conv1D expects (N,C_in,L_in), where C_in=number of channels=measurement dimension, L_in=length of signal sequence
+        # Y_test = torch.ones(n_samples_test).float().numpy()
+
+    elif H.dataset in list(
```

Comment:
Double no! To check whether a key exist in a dictionary, you can just do `key in dct`. There is no need to do `key in dct.keys()`.

But never call `list` on it, since that turns the lookup from `O(1)` to `O(n)`.

Reply:
Thanks! Done.